plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/TrinoFileSystemCache.java

```diff
@@ -94,11 +114,13 @@ private synchronized FileSystem getInternal(URI uri, Configuration conf, long un
         if (fileSystemHolder == null) {
             int maxSize = conf.getInt("fs.cache.max-size", 1000);
             if (map.size() >= maxSize) {
+                stats.newGetCallFailed();
```

Comment:
Can it fail only here?

or, can `FileSystem fileSystem = createFileSystem(uri, conf);` below fail too?

Reply:
Yes, you are right: `createFileSystem` can fail if an invalid filesystem scheme is used.

I will add a call to increment the statistics in that method too.